.github/scripts/upload_tests_results.py

```diff
@@ -0,0 +1,342 @@
+                    "pull": pull,
+                    "run_time": run_time,
+                    "job_name": job_name,
+                    "job_id": job_id,
+                    "suite_name": suite_name,
+                    "raw_test_name": rawname,
+                    "file": filename,
+                    "test_name": testname,
+                    "fixture": fixture,
+                    "time": float(time),
+                    "status": status,
+                    "status_description": status_description.replace("\r\n", ";;").replace("\n", ";;").replace("\"", "'"),
+                    "log": "" if testcase.find("properties/property/[@name='url:log']") is None else testcase.find("properties/property/[@name='url:log']").get('value'),
+                    "logsdir": "" if testcase.find("properties/property/[@name='url:logsdir']") is None else testcase.find("properties/property/[@name='url:logsdir']").get('value'),
+                    "stderr": "" if testcase.find("properties/property/[@name='url:stderr']") is None else testcase.find("properties/property/[@name='url:stderr']").get('value'),
+                    "stdout": "" if testcase.find("properties/property/[@name='url:stdout']") is None else testcase.find("properties/property/[@name='url:stdout']").get('value'),
+
+                }
+            )
+    return results
+
+
+def get_codeowners_for_tests(entries, tests_data):
+    tests_data_with_owners = []
+    for test in tests_data:
+        owners = []
+        target_path = test["suite_name"]
+        for path, path_owners in entries:
+            path = path + "/" if path[-1] != "/" else path
+            path = path + "*" if path[-1] != "*" else path
+            if fnmatch.fnmatch(f"/{target_path}", path):
+                owners.extend(path_owners)
+
+        test["owners"] = ",".join(owners)
+        tests_data_with_owners.append(test)
+    return tests_data_with_owners
+
+
+def upload_codeowners(session, entries):
+    # Вставка данных в таблицу
+    for entry in entries:
+        path, owners = entry
+        sql = f"""
+        UPSERT INTO codeowners (path, owners) VALUES 
+            ("{path}", {str(owners).replace("'", '"')});
+        """
+        session.transaction().execute(sql, commit_tx=True)
+
+
+def create_table(session, sql):
+
+    try:
+        session.execute_scheme(sql)
+    except ydb.issues.AlreadyExists:
+        pass
+    except Exception as e:
+        print(f"Error creating table: {e}, sql:\n{sql}", file=sys.stderr)
+        raise e
+
+
+# Создание таблицы
+def create_codeowners_table(session, table_path):
+    sql = f"""
+    --!syntax_v1
+    CREATE TABLE IF NOT EXISTS `{table_path}` (
+        path Utf8,
+        owners Utf8,
+        PRIMARY KEY (path)
+    );
+    """
+    create_table(session, sql)
+
+
+def create_tests_table(session, table_path):
+    # Создание таблицы, если ее еще нет
+    sql = f"""
+    --!syntax_v1
+    CREATE TABLE IF NOT EXISTS `{table_path}` (
+        build_type Utf8,
```

Comment:
Let make separate table for workflow runs and extract there such things as build_type, sha, job_name etc 

It would be handy for other tables

Reply:
m agree about separete table for some workflow data, but
- I can't figure it out how to separate it better
- datalens will blows up ( or performance become critical poor)  if on every view (grid/table/ etc) it should make joins of all history if runs with workflow data before aggregation and filtrations. As far as I understand how datalens works

Reply:
Ok